crates/optimism/node/src/txpool.rs

```diff
@@ -58,14 +58,25 @@ impl OpPooledTransaction {
     }
 
     /// Returns the estimated compressed size of a transaction in bytes scaled by 1e6.
-    // This value is computed based on the following formula:
-    // `max(minTransactionSize, intercept + fastlzCoef*fastlzSize)`
+    /// This value is computed based on the following formula:
+    /// `max(minTransactionSize, intercept + fastlzCoef*fastlzSize)`
     pub fn estimated_compressed_size(&self) -> u32 {
-        // TODO(mattsse): use standalone flz compute function
-        *self.estimated_tx_compressed_size.get_or_init(|| self.inner.encoded_length as u32)
+        *self
+            .estimated_tx_compressed_size
+            .get_or_init(|| tx_estimated_size_fjord(&self.inner.transaction().encoded_2718()))
     }
 }
 
+/// Calculate the estimated compressed transaction size in bytes, scaled by 1e6.
+/// This value is computed based on the following formula:
+/// max(minTransactionSize, intercept + fastlzCoef*fastlzSize)
+// TODO(mattsse): replace with library fn from revm or maili once available
+fn tx_estimated_size_fjord(input: &[u8]) -> u32 {
+    let fastlz_size = maili_flz::flz_compress_len(input);
+
+    fastlz_size.saturating_mul(836_500).saturating_sub(42_585_600).max(100_000_000)
```

Comment:
actually no sorry 100M is the minimum size, in theory a large enough input returns N where N * 836_500 > u32::MAX, so it should either be impossible or this should be made to use u64